pkg/asset/installconfig/baremetal/baremetal.go

```diff
@@ -0,0 +1,142 @@
+// Package baremetal collects bare metal specific configuration.
+package baremetal
+
+import (
+	"fmt"
+	"gopkg.in/AlecAivazis/survey.v1"
+
+	"github.com/openshift/installer/pkg/types/baremetal"
+	baremetaldefaults "github.com/openshift/installer/pkg/types/baremetal/defaults"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+// Platform collects bare metal specific configuration.
+func Platform() (*baremetal.Platform, error) {
+	var externalBridge, provisioningBridge, provisioningNetworkInterface, apiVIP, ingressVIP string
+	var hosts []*baremetal.Host
+
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "External bridge",
+				Help:    "External bridge is used for external communication.",
+				Default: baremetaldefaults.ExternalBridge,
+			},
+		},
+	}, &externalBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning bridge",
+				Help:    "Provisioning bridge is used to provision machines.",
+				Default: baremetaldefaults.ProvisioningBridge,
+			},
+		},
+	}, &provisioningBridge)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning Network Interface",
+				Help:    "ProvisioningNetworkInterface is the name of the network interface on a control plane host connected to the provisioning network.",
+			},
+			Validate: survey.Required,
+		},
+	}, &provisioningNetworkInterface); err != nil {
+		return nil, err
+	}
+
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "API VIP",
+				Help:    "The VIP to be used for internal API communication. If blank, the address will be looked up from DNS.",
+				Default: baremetaldefaults.APIVIP,
```

Comment:
For API and Ingress, we should try to detect them.  Maybe extract this stuff here to a helper method: https://github.com/openshift/installer/blob/master/pkg/types/baremetal/defaults/platform.go#L79-L99

Reply:
Added pretty much all of the platform. PTAL.

Reply:
Oh that stinks :man_facepalming:

Reply:
No longer relevant because we dropped VIPs.